xxxApp/src/Makefile

```diff
@@ -23,422 +27,385 @@ PROD_IOC = xxx
 # Custom stuff in development
 
 ##################################################################
-# First, define xxx_Common_DBD and xxx_Common_LIBS from the modules
+# First, define $(DBD_NAME)_DBD and $(PROD_NAME)_LIBS from the modules
 # defined in RELEASE
 ##################################################################
-DBD += xxx_Common.dbd
-xxx_Common_DBD += base.dbd
+ifeq (win32-x86, $(findstring win32-x86, $(T_A)))
+  DBD_NAME  = $(DBD_PREFIX)Win32
+  WINDOWS_TARGET = YES
+endif
+ifeq (windows-x64, $(findstring windows-x64, $(T_A)))
+  DBD_NAME  = $(DBD_PREFIX)Win64
+  WINDOWS_TARGET = YES
+endif
+
+ifeq (vxWorks,$(findstring vxWorks, $(T_A)))
+  DBD_NAME  = $(DBD_PREFIX)VX
+endif
+
+ifeq (linux,$(findstring linux, $(T_A)))
```

Comment:
Hi Mark,

These changes look very sensible to me. I would suggest using `OS_CLASS` for detecting VxWorks and Linux (like you do with Darwin lower down) as some sites add their own targets with arch-names that don't include the full OS name, e.g. `rhel6-amd64`.

You could probably shorten this whole block to something like (not tested):
```make
DBD_SUFFIX_cygwin32 = $(error Cygwin not supported)
DBD_SUFFIX_Darwin = Darwin
DBD_SUFFIX_Linux = Linux
DBD_SUFFIX_RTEMS = $(error RTEMS not supported)
DBD_SUFFIX_vxWorks = Vx
DBD_SUFFIX_WIN32 = Win$(if $(findstring 32, $(T_A)),32,64)
DBD_NAME = $(DBD_PREFIX)$(DBD_SUFFIX_$(OS_CLASS))
```
If `WINDOWS_TARGET` is only used in this Makefile, maybe replace the `ifdef WINDOWS_TARGET` statements with `ifeq ($(OS_CLASS), WIN32)` to avoid creating another config variable?

- Andrew